internal/healthcheck/healthcheck.go

```diff
@@ -90,8 +89,24 @@ func (c *Check) HandleReadiness(w http.ResponseWriter, _ *http.Request) {
 		return
 	}
 
-	err := c.proxy.CheckConnections(ctx)
-	if err != nil && !ready(err, c.minReady) {
+	var minReady int
+	q := req.URL.Query()
+	if v := q.Get("min-ready"); v != "" {
+		n, err := strconv.Atoi(v)
+		// If the query value parsed correctly, set it to minReady. Otherwise,
+		// ignore it.
+		if err == nil {
+			minReady = n
+		}
+	}
+
+	n, err := c.proxy.CheckConnections(ctx)
+	if minReady > n {
+		c.logger.Errorf("[Health Check] min-ready was greater than registered instances")
```

Comment:
This was meant to be a validation of input in case a user provided a value of min-ready greater than the number of registered instances. But its value seems low and will be handled by `ready`, so I'll remove it.